apps/state/lib/state/routes_by_service.ex

```diff
@@ -0,0 +1,114 @@
+defmodule State.RoutesByService do
+  @moduledoc """
+  Allows finding `Model.Route.t` that are active on a given date.
+  """
+  use Timex
+  require Logger
+  use Events.Server
+  import Events
+  import State.Helpers
+
+  @table __MODULE__
+
+  def start_link do
+    GenServer.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  def size do
+    safe_ets_size(@table)
+  end
+
+  def for_service_id(service_id) do
+    @table |> :ets.match_object({{service_id, :_}, :_}) |> do_get_routes()
+  end
+
+  def for_service_id_and_types(service_id, route_types) do
+    @table
+    |> :ets.select(
+      for(
+        type <- route_types,
+        do: {{{:"$0", :"$1"}, :"$2"}, [{:==, :"$0", service_id}, {:==, :"$1", type}], [:"$_"]}
```

Comment:
You don't need the guards here: `{{service_id, type}, :_}, [], [:"$_"]}` should work fine.